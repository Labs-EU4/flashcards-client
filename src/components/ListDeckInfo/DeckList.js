import React, {useEffect, useState} from "react";
import {axiosWithAuth} from "../../utils/axios";
import DeckCard from "./DeckCard";

export default function DecksList({requestAddrs}) {
  const [decks, setDecks] = useState([]);

  useEffect(() => {
    axiosWithAuth()
      .get(requestAddrs)
      .then(res => {
        console.log(res);
        // setDecks(
        //   res.data.data.filter(deck => {
        //     return deck.flashcards[0] !== null;
        //   })
        // );
        setDecks(res.data.data);
      })
      .catch(err => {
        console.log(err);
      });
  }, [requestAddrs]);

  return (
    <div>
      {/* {console.log(decks)}
      {decks.map(deck => {
        if (deck.flashcards.length >= 1) {
          return <DeckCard deck={deck} />;
        }
      })} */}
      {decks.length > 0 ? (
        decks.map(deck => {
          return <DeckCard deck={deck} />;
        })
      ) : (
        <h2>You have no decks right now</h2>
      )}
    </div>
  );
}
