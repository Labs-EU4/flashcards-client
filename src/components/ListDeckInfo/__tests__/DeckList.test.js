import React from "react";
import * as rtl from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import DeckList from "../DeckList";

const NoDecksMessage = () => {
  return wrapper.queryByText("You have no decks right now");
};

const DeckCard = () => {
  return wrapper.queryByTestId("deck_card_container");
};

jest.mock("../../../utils/axios", () => {
  return {
    axiosWithAuth: jest.fn(() => {
      return {
        get: jest.fn().mockResolvedValue([
          {
            deck_name: "Test Deck",
            author: "John Doe",
            flashcards: ["just", "mock", "data"],
          },
        ]),
      };
    }),
  };
});

let wrapper;
beforeEach(() => {
  wrapper = rtl.render(<DeckList requestAddrs="/test" />);
});

it("renders without crashing", () => {
  expect(wrapper.container).toMatchSnapshot();
  expect(NoDecksMessage()).toBeInTheDocument();
  //   expect(await rtl.waitForElement(() => DeckCard())).toBeInTheDocument();
});
