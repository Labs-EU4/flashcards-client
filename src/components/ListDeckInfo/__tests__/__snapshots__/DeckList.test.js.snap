// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div>
    <h2>
      You have no decks right now
    </h2>
  </div>
</div>
`;
